In [ ]:
import numpy as np
import pandas as pd

: 

Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [2]:
arr = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
arr1 = np.sort(arr)

In [3]:
print('несмещенная:', np.std(arr1, ddof=1)) #СКО
print('смещенная:', np.std(arr1, ddof=0))

несмещенная: 31.62460734101981
смещенная: 30.823854398825592


элемент минус среднее занчение 100-65.3 и взоводим в квадрат
1204.09 216.09 94.09 136.89 561.69 1043.29 412.09 1624.09 0.09 2332.89 1246.09 1705.69 68.89 22.09 94.09 0.09 349.69 610.09 7174.09 = 18896.11 / на количество элементов 20 и взять корень квадратный, = 30.823

In [4]:
print('несмещенная:', arr.var(ddof=1)) # дисперсия
print('несмещенная:', np.var(arr1, ddof=1))
print('смещенная:', np.var(arr1, ddof=0))
print('смещенная:', np.var(arr, ddof=0))

несмещенная: 1000.1157894736842
несмещенная: 1000.115789473684
смещенная: 950.1099999999999
смещенная: 950.11


элемент минус среднее занчение 100-65.3 и взоводим в квадрат
1204.09 216.09 94.09 136.89 561.69 1043.29 412.09 1624.09 0.09 2332.89 1246.09 1705.69 68.89 22.09 94.09 0.09 349.69 610.09 7174.09 = 18896.11 / на количество элементов 20 (смещеная) = 950.11; (n-1) 19 элементов (несмещеная) = 1000.115

In [5]:
np.mean(arr1)

65.3

среднее арифметическое = сумма всех элементов 1306 / на число элементов 20 = 65.3

In [6]:
len(arr)

20

В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

In [7]:
import math as mt

In [8]:

def combination (n, k):
    return mt.factorial(n) // (mt.factorial(k) * mt.factorial(n - k ))

In [9]:
a = combination (3, 2)/combination (8, 2) * combination (5, 3) * combination (7, 1)/combination (12, 4)
b = combination (5, 1)* combination(3, 1)/combination (8, 2) * combination (5, 2) * combination (7, 2)/combination (12, 4) 
c = combination (5, 2)/combination (8, 2) * combination (5, 1) * combination (7, 3)/combination (12, 4)
a + b + c

0.3686868686868686

На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

P(B|A) = P(A|B) * P(B) // P(A) ; P(B1,2,3) = 1/3; P(A) = P(A|B1) * P(B1) + P(A|B2) * P(B2) + P(A|B3) * P(B3)

In [10]:
pa = 0.9 * 1/3 + 0.8 * 1/3+ 0.6 * 1/3
pb = 1/3
p1 = 0.9 * pb / pa
p2 = 0.8 * pb/ pa
p3 = 0.6 * pb/ pa
p1, p2, p3

(0.391304347826087, 0.3478260869565218, 0.2608695652173913)

В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

P(B|A) = P(A|B) * P(B) // P(A) ; P(B) = 1; P(B1) = 1/4; P(B2) = 1/4; P(B3) = 1/2 

In [11]:
pa = 0.8 * 1/4 + 0.7 * 1/4+ 0.9 * 1/2
pb1 = 1/4
pb2 = 1/4
pb3 = 1/2
p1 = 0.8 * pb1 / pa
p2 = 0.7 * pb2 / pa
p3 = 0.9 * pb3 / pa
p1, p2, p3

(0.24242424242424246, 0.21212121212121213, 0.5454545454545455)

Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?

P(D) = P(A1*A2*A3) = P(A1) * P(A2) * P(A3)

p(E) = P(A1*A2*_A3) + P(A1*_A2*A3) + P(_A1*A2*A3)

P(F) = (P(A1*_A2*_A3)+P(_A1*A2*_A3)+P(_A1*_A2*A3) + (P(A1*A2*_A3) + P(A1*_A2*A3) + P(_A1*A2*A3) + P(A1*A2*A3))

P(G) = (P(A1*_A2*_A3)+P(_A1*A2*_A3)+P(_A1*_A2*A3) + (P(A1*A2*_A3) + P(A1*_A2*A3) + P(_A1*A2*A3))

In [12]:

pa1 = 0.1
pa2 = 0.2
pa3 = 0.25

pd = pa1 * pa2 * pa3
pe = (pa1 * pa2 * (1-pa3)) + ((1-pa1) * pa2 * pa3) + (pa1 * (1-pa2) * pa3)
pf = (pa1 * (1-pa2) * (1-pa3)) + ((1-pa1) * (1-pa2) * pa3) + ((1-pa1 )* pa2 * (1-pa3)) + (pe + pd)
pf1 = 1-((1-pa1) * (1-pa2) * (1-pa3))
pg =  (pa1 * (1-pa2) * (1-pa3)) + ((1-pa1) * (1-pa2) * pa3) + ((1-pa1 )* pa2 * (1-pa3)) + pe

print(pd)
print(pe)
print(pf1)
print(pf)
print(pg)

0.005000000000000001
0.08000000000000002
0.45999999999999996
0.4600000000000001
0.45500000000000007
